spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/webservices/client/WebServiceTemplateExchangeTags.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.boot.actuate.metrics.webservices.client;
+
+import io.micrometer.core.instrument.Tag;
+
+import org.springframework.util.StringUtils;
+import org.springframework.ws.WebServiceMessage;
+import org.springframework.ws.client.WebServiceClientException;
+import org.springframework.ws.client.WebServiceIOException;
+import org.springframework.ws.context.MessageContext;
+import org.springframework.ws.soap.SoapMessage;
+import org.springframework.ws.transport.WebServiceConnection;
+import org.springframework.ws.transport.context.TransportContext;
+import org.springframework.ws.transport.context.TransportContextHolder;
+
+/**
+ * Factory methods for creating {@link Tag Tags} related to a soap request-response
+ * exchange performed by a {@link org.springframework.ws.client.core.WebServiceTemplate}.
+ *
+ * @author Dmytro Nosan
+ * @since 2.1.0
+ **/
+public abstract class WebServiceTemplateExchangeTags {
+
+	private static final Tag OK = Tag.of("status", "OK");
+
+	private static final Tag IO_ERROR = Tag.of("status", "IO_ERROR");
+
+	private static final Tag CLIENT_ERROR = Tag.of("status", "CLIENT_ERROR");
+
+	private static final Tag SERVER_ERROR = Tag.of("status", "SERVER_ERROR");
+
+	private static final Tag NO_URI = Tag.of("uri", "NONE");
+
+	private static final Tag NO_ACTION = Tag.of("action", "NONE");
+
+	/**
+	 * Creates a {@code Tag(uri)} using {@link TransportContext#getConnection()}.
+	 * @return the uri tag
+	 */
+	public static Tag uri() {
+		String uri = getUri();
+		return (!StringUtils.hasText(uri) ? NO_URI : Tag.of("uri", uri));
+	}
+
+	/**
+	 * Creates a {@code Tag(action)} using {@link SoapMessage#getSoapAction()}.
+	 * @param messageContext Context of the SOAP message.
+	 * @return the action tag
+	 */
+	public static Tag action(MessageContext messageContext) {
```

Comment:
From the spec, it appears that action could potentially be a high cardinality tag:

![image](https://user-images.githubusercontent.com/1697736/40988408-96a7bae0-68b0-11e8-91e7-dd24f5108dcc.png)

Reply:
I'm not sure about `SOAPAction: "myapp.sdl" -> myapp.sdl`
@jkschneider Should we do something with dots?

Reply:
Don't concern yourself with dots. If they aren't an acceptable character for a particular monitoring system, it is the responsibility of Micrometer's `NamingConvention` for that registry to strip or escape them.

Reply:
that was my question, ty :)